easybuild/easyblocks/e/easybuildmeta.py

```diff
@@ -73,6 +73,10 @@ def __init__(self, *args, **kwargs):
             self.easybuild_pkgs.extend(['vsc-base', 'vsc-install'])
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
+        elif (LooseVersion(self.version) >= LooseVersion('5.0') and
+              any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources'])):
```

Comment:
`source` may not be a dict, depends on the easyconfig file being used, see `EasyBuild-4.9.1.eb` for example, so we need be be more careful here: https://github.com/branfosj/easybuild-easyblocks/pull/2